src/accelerate/accelerator.py

```diff
@@ -2753,7 +2670,7 @@ def _inner(folder):
         # Save the optimizers taking care of FSDP and DeepSpeed nuances
         optimizers = []
         if self.distributed_type == DistributedType.FSDP:
-            for opt in self._optimizers:
+            for i, opt in enumerate(self._optimizers):
```

Comment:
Was this a bug in the old code?

Reply:
Yes, this was brought to our notice via https://github.com/huggingface/accelerate/issues/2175